server/routes/public.js

```diff
@@ -29,15 +29,48 @@ router.get('/api/login', async (req, res) => {
 });
 
 router.get('/api/create-user', async (req, res) => {
-    const firstName = req.query.firstName;
-    const lastName = req.query.lastName;
-    const type = req.query.type; //note: 'type' is called 'role' in the UI, but refers to either a worker or a student.
-    const studentID = req.query.studentID;
-    const email = req.query.email;
-    const password = req.query.password;
-    const phone = req.query.phone;
-
-    const user = await accountHandler.createUserAccount(firstName, lastName, type, studentID, email, password, phone);
+    // joi verification
+    // Validate appropriate parameters are passed into the create account endpoint.
+    const paramSchema = Joi.object({
+       
+       //The following joi verifications make sure the values inputted adhere to the format requirements
+       //specified in the migrations> create-worker-table and migrations> create-student-table.
+
+        email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net', 'ca'] } }).max(320).required(),
+        password: Joi.string().min(3).max(40).required(),
+        phone: Joi.string().max(20).phoneNumber(),
+        first_name: Joi.string().min(1).max(20).required(),
+        last_name: Joi.string().min(1).max(20).required(),
+        student_id: Joi.number().positive().integer(),
+        worker_id: Joi.number().positive().integer(),
+        userType: Joi.string().min(1).valid(['student', 'worker']),
+        specialization: Joi.string().valid(['social worker', 'guidance councellor']),//verify options
+        access_token: Joi.string().alphanum().min(3).max(30).required(), 
+        //insert joi verification for first_name,last_name, type,student_id,workerID,phone
+    });
+    // the endpoint `/api/create-user` accepts ALL parameters for both user types, 
+    //as listed in the migrations> create-worker-table and migrations> create-student-table
+
+    const paramFirst_name = req.query.first_name;
```

Comment:
Hi Nancy,
I would recommend using lowerCamelCase instead of underscores in the variable naming to maintain code consistency. For example, `paramFirst_name` would be changed to `paramFirstName`

Reply:
Hi Amy,
Thank you for pointing that for me!
My action: I've changed the const param_names to be in camelCase. Additionally, I have also made these changes to the `accountHandler.createUserAccount()` and the `paramSchema.validate()`

You can now see these changes in:
lines 55-64 of `server/routes/public.js`
line 70  of `server/routes/public.js`
line 74  of `server/routes/public.js`
line 85  of `server/routes/public.js`

Please confirm if this takes care of your suggestion!

Reply:
Hi Nancy, 
Yes, that's perfect. Thank you for making those changes!